src/simplewallet/simplewallet.cpp

```diff
@@ -465,6 +476,587 @@ bool simple_wallet::print_fee_info(const std::vector<std::string> &args/* = std:
+      {
+        message_writer() << tr("Duplicate key found, ignoring");
+        secret_keys[j] = secret_keys.back();
+        public_keys[j] = public_keys.back();
+        secret_keys.pop_back();
+        public_keys.pop_back();
+        --j;
+      }
+    }
+  }
+
+  // people may include their own, weed it out
+  crypto::hash hash;
+  crypto::cn_fast_hash(&m_wallet->get_account().get_keys().m_view_secret_key, sizeof(crypto::secret_key), hash);
+  for (size_t i = 0; i < secret_keys.size(); ++i)
+  {
+    if (rct::sk2rct(secret_keys[i]) == rct::hash2rct(hash))
+    {
+      message_writer() << tr("Local key is present, ignoring");
+      secret_keys[i] = secret_keys.back();
+      public_keys[i] = public_keys.back();
+      secret_keys.pop_back();
+      public_keys.pop_back();
+      --i;
+    }
+    else if (public_keys[i] == m_wallet->get_account().get_keys().m_account_address.m_spend_public_key)
+    {
+      fail_msg_writer() << tr("Found local spend public key, but not local view secret key - something very weird");
+      return true;
+    }
+  }
+
+  LOCK_IDLE_SCOPE();
+
+  try
+  {
+    std::string multisig_extra_info = m_wallet->make_multisig(orig_pwd_container->password(), secret_keys, public_keys, threshold);
+    if (!multisig_extra_info.empty())
+    {
+      success_msg_writer() << tr("Another step is needed");
+      success_msg_writer() << multisig_extra_info;
+      success_msg_writer() << tr("Send this multisig info to all other participants, then use finalize_multisig <info1> [<info2>...] with others' multisig info");
+      return true;
+    }
+  }
+  catch (const std::exception &e)
+  {
+    fail_msg_writer() << tr("Error creating multisig: ") << e.what();
+    return true;
+  }
+
+  uint32_t total = secret_keys.size() + 1;
+  success_msg_writer() << std::to_string(threshold) << "/" << total << tr(" multisig address: ")
+      << m_wallet->get_account().get_public_address_str(m_wallet->testnet());
+
+  return true;
+}
+
+bool simple_wallet::finalize_multisig(const std::vector<std::string> &args)
+{
+  bool ready;
+  if (!m_wallet->multisig(&ready))
+  {
+    fail_msg_writer() << tr("This wallet is not multisig");
+    return true;
+  }
+  if (ready)
+  {
+    fail_msg_writer() << tr("This wallet is already finalized");
+    return true;
+  }
+
+  const auto orig_pwd_container = get_and_verify_password();
+  if(orig_pwd_container == boost::none)
+  {
+    fail_msg_writer() << tr("Your original password was incorrect.");
+    return true;
+  }
+
+  if (args.size() < 2)
+  {
+    fail_msg_writer() << tr("usage: finalize_multisig <multisiginfo1> [<multisiginfo2>...]");
+    return true;
+  }
+
+  // parse all multisig info
+  std::unordered_set<crypto::public_key> public_keys;
+  std::vector<crypto::public_key> signers(args.size(), crypto::null_pkey);
+  for (size_t i = 0; i < args.size(); ++i)
+  {
+    if (!m_wallet->verify_extra_multisig_info(args[i], public_keys, signers[i]))
```

Comment:
AIUI, we expect the number of `<multisiginfo_N>` arguments to be the same as the total number of participants, which is encoded in each extra multisig info file. However, currently there's no error checking when this assumption isn't met, which I think is not user friendly.

Reply:
Yes, this is a bit crap, as the way to get a string error is via exceptions :/ Fixing.

Reply:
Note that the fix is in the bit of code below, wihch was rejecting bad number of signers (but silently).